### PR TITLE
Store compact view status

### PR DESCRIPTION
**Pull request by thomtrp**
Currently, compact view status is only set in the state. This is not stored in DB so it gets reset when the page is refreshed.
This PR stores it as a boolean in the view.

https://github.com/twentyhq/twenty/assets/22936103/62f4f98a-b464-43f6-9352-f8d05952d673

### CODE REVIEW

File: packages/twenty-front/src/modules/object-record/record-index/components/RecordIndexBoardContainerEffect.tsx [hunk 1]
```diff
@@ -33,6 +34,15 @@ export const RecordIndexBoardContainerEffect = ({
     getOnFetchMoreVisibilityChangeState,
   } = useRecordBoard(recordBoardId);
 
+  const { currentViewSelector } = useViewScopedStates({
+    viewScopeId: viewBarId,
```
Comment (thomtrp) [inline packages/twenty-front/src/modules/object-record/record-index/components/RecordIndexBoardContainerEffect.tsx:38]: this is used that way in other places in the code. Should I do a refacto?

File: packages/twenty-server/src/workspace/workspace-sync-metadata/standard-objects/view.object-metadata.ts [hunk 2]
```diff
@@ -41,6 +41,14 @@ export class ViewObjectMetadata extends BaseObjectMetadata {
   })
   type: string;
 
+  @FieldMetadata({
+    type: FieldMetadataType.BOOLEAN,
+    label: 'Compact View',
+    description: 'Describes if the view is in compact mode',
+    defaultValue: { value: false },
+  })
+  isCompactView: boolean;
```
Comment (Weiko) [inline packages/twenty-server/src/workspace/workspace-sync-metadata/standard-objects/view.object-metadata.ts:50]: isCompact?

Comment (thomtrp) [reply packages/twenty-server/src/workspace/workspace-sync-metadata/standard-objects/view.object-metadata.ts]: @Weiko updated

**Review by Weiko — COMMENTED** [review]
Backend lgtm 😄

File: packages/twenty-front/src/modules/object-record/record-index/components/RecordIndexBoardContainerEffect.tsx (hunk 1 above)
Comment (charlesBochet) [reply packages/twenty-front/src/modules/object-record/record-index/components/RecordIndexBoardContainerEffect.tsx:38]: No, we'll take care of that while refactoring the ViewBar later this month

File: packages/twenty-front/src/modules/object-record/record-index/components/RecordIndexBoardContainerEffect.tsx [hunk 3]
```diff
@@ -98,5 +108,14 @@ export const RecordIndexBoardContainerEffect = ({
     setContextMenuEntries?.();
   }, [setActionBarEntries, setContextMenuEntries]);
 
+  useEffect(() => {
+    if (currentView?.isCompact !== isCompactModeActive) {
+      setIsCompactModeActive(!!currentView?.isCompact);
+    }
+    // We don't want to re-run this effect when the state changes.
+    // We only want to make sure that the state reflects the current view.
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```
Comment (charlesBochet) [inline packages/twenty-front/src/modules/object-record/record-index/components/RecordIndexBoardContainerEffect.tsx:117]: We should not try to block react re-rendering. It's an effect, it is not a problem :)

**Review by charlesBochet — APPROVED** [review]

**Review by charlesBochet — APPROVED** [review]

**Pull request merged**